### ISSUE TITLE
[Bug]: Can't seem to get templating working

### ISSUE THREAD

**Issue by polyphilz** [stepci/stepci#177]
### What happened?

I ran something along the lines of:

```
stepci run workflow.yml -s token=$(<script that generates token>)
```

...and in `workflow.yml`:

```
version: '1.1'
name: GetCompaniesTest
components:
  credentials:
    ops:
      bearer:
        token: ${{secrets.token}}
env:
  host: localhost:3001/v1
tests:
  example:
    steps:
      - name: GET request
        http:
          url: http://${{env.host}}/companies
          method: GET
          auth:
            bearer:
              $ref: '#/components/credentials/ops'
          check:
            status: /^20/
```

### What did you expect to happen?

I would expect the token to be passed to the workflow but instead this is what I am getting:

```
 FAIL  example ⏲ 0.014s ⬆ 0 bytes ⬇ 0 bytes                                                                                                                                                                            
                                                                                                                                                                                                                       
Summary                                                                                                                                                                                                                
                                                                                                                                                                                                                       
  ✕ GET request failed after 0.014s                                                                                                                                                                                    
                                                                                                                                                                                                                       
● example › GET request                                                                                                                                                                                                
                                                                                                                                                                                                                       
Request  HTTP                                                                                                                                                                                                          
                                                                                                                                                                                                                       
GET http://localhost:3001/v1/companies HTTP/1.1                                                                                                                                                                        
Authorization: Bearer undefined                                                                                                                                                                                        

Response

HTTP/1.1 401 Unauthorized
...
```

The bearer token is `undefined`. Am I doing something wrong when passing this secret, or in the templating?

### Version

2.7.0

### Environment

v20.8.0

### How can we reproduce this bug?

Workflow pasted above

### Relevant log output

```shell
FAIL  example ⏲ 0.014s ⬆ 0 bytes ⬇ 0 bytes                                                                                                                                                                            
                                                                                                                                                                                                                       
Summary                                                                                                                                                                                                                
                                                                                                                                                                                                                       
  ✕ GET request failed after 0.014s                                                                                                                                                                                    
                                                                                                                                                                                                                       
● example › GET request                                                                                                                                                                                                
                                                                                                                                                                                                                       
Request  HTTP                                                                                                                                                                                                          
                                                                                                                                                                                                                       
GET http://localhost:3001/v1/companies HTTP/1.1                                                                                                                                                                        
Authorization: Bearer undefined                                                                                                                                                                                        

Response

HTTP/1.1 401 Unauthorized
x-powered-by: Express
access-control-allow-origin: *
content-type: application/json; charset=utf-8
content-length: 161
date: Mon 06 Nov 2023 00:01:24 GMT
connection: keep-alive
keep-alive: timeout=5

{"response":{"statusCode":401,"message":"Unauthorized","error":"Unauthorized"},"status":401,"options":{},"message":"Unauthorized","name":"UnauthorizedException"}

Checks

Status

  ✕ 401 (expected /^20/)

Tests: 1 failed, 0 passed, 1 total
Steps: 1 failed, 0 skipped, 0 passed, 1 total
Time:  0.241s, estimated 0s
CO2:   0.00006g

Workflow failed after 0.241s
Give us your feedback on https://step.ci/feedback
```


### Would you be interested in working on a bugfix for this issue?

- [X] Yes! Assign me

**Comment by mishushakov** [comment]
Hey, you can do either:


```yaml
version: '1.1'
name: GetCompaniesTest
env:
  host: localhost:3001/v1
tests:
  example:
    steps:
      - name: GET request
        http:
          url: http://${{env.host}}/companies
          method: GET
          auth:
            bearer:
              token: ${{secrets.token}}
          check:
            status: /^20/
```

or

```yaml
version: '1.1'
name: GetCompaniesTest
components:
  credentials:
    ops:
      bearer:
        token: ${{secrets.token}}
env:
  host: localhost:3001/v1
tests:
  example:
    steps:
      - name: GET request
        http:
          url: http://${{env.host}}/companies
          method: GET
          auth:
            $ref: '#/components/credentials/ops'
          check:
            status: /^20/
```

**Comment by polyphilz** [comment]
Unfortunately for both of those, I am getting that the bearer token is `undefined` when passed as:

```
stepci run workflow.yml -s token=$(<script that generates token>)
```

**Comment by polyphilz** [comment]
Weird, just tried again with the second and it's looking fine now, but still can't get it working with the reusable components method.

**Comment by mishushakov** [comment]
I have pushed a fix in a new runner version (1.10.10). You can just reinstall stepci package and it should be working with reusable components as you expected!

Feel free to reopen if the fix hasn't worked.
